src/common.mak

```diff
@@ -0,0 +1,204 @@
+# Common Makefile variables (shared between DMD, DRuntime and Phobos)
+# Be careful when updating
+# This file expects the following variables to be defined in the including Makefile:
+# - D_HOME        Root of all D repositories
+
+###############################################################################
+# Common directories
+################################################################################
+
+DMD_DIR = $(D_HOME)/dmd
+DRUNTIME_DIR = $(D_HOME)/druntime
+PHOBOS_DIR = $(D_HOME)/phobos
+DLANG_ORG_DIR = $(D_HOME)/dlang.org
+TOOLS_DIR = $(D_HOME)/tools
+INSTALL_DIR = $(D_HOME)/install
+INSTALLER_DIR = $(D_HOME)/installer
+TMP?=/tmp
+GIT_HOME=https://github.com/dlang
+
+# get OS and MODEL
+include $(D_HOME)/dmd/src/osmodel.mak
+
+###############################################################################
+# Common variables
+################################################################################
+
+# Set VERSION, where the file is that contains the version string
+VERSION=$(DMD_DIR)/VERSION
+MAKEFILE = $(firstword $(MAKEFILE_LIST))
+
+################################################################################
+# Output directories
+################################################################################
+
+DMD_GENERATED=$(DMD_DIR)/generated
+DRUNTIME_GENERATED=$(DRUNTIME_DIR)/generated
+PHOBOS_GENERATED=$(PHOBOS_DIR)/generated
```

Comment:
I wasn't sure whether these should be named `PHOBOS_GENERATED` or `PHOBOS_GENERATED_DIR`.
I think for consistency it would be better to stick the `_DIR` convention. Opinions?

Reply:
These don't seem common - on the contrary, they're specific to each project. Do we need them here for crosstalk?

Reply:
Yes we do. By defining them once here, we don't need to redefine this over and over again whenever we want to access e.g. the generated DMD binary or Phobos library